### ISSUE TITLE
[libiconv:x64-windows] build failure

### ISSUE THREAD

**Issue by t-sharp** [microsoft/vcpkg#17305]
**Host Environment**
- OS: Microsoft Windows 10 Pro Version 20H2 (Build 19042.928)
- Compiler: Visual Studio 2019 Version 16.9.4

**To Reproduce**
`./vcpkg.exe install libiconv:x64-windows`

**Failure logs** 
Output from PowerShell:
```
Computing installation plan...
The following packages will be built and installed:
    libiconv[core]:x64-windows -> 1.16#8
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\Tobias\AppData\Local\vcpkg\archives\26\26d2167072fc54a6fbc6f1af5a41842c0ebde6d6.zip
Starting package 1/1: libiconv:x64-windows
Building package libiconv[core]:x64-windows...
-- Using cached U:/Developer/vcpkg/downloads/libiconv-1.16.tar.gz
-- Cleaning sources at U:/Developer/vcpkg/buildtrees/libiconv/src/1.16-de9101a181.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source U:/Developer/vcpkg/downloads/libiconv-1.16.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Using source at U:/Developer/vcpkg/buildtrees/libiconv/src/1.16-de9101a181.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" -E tar xzf U:/Developer/vcpkg/downloads/msys-binutils-2.34-4-x86_64.pkg.tar.zst
    Working Directory: U:/Developer/vcpkg/downloads/tools/msys2/d589ebc2087b2b2c.tmp
    Error code: 1
    See logs for more information:
      U:\Developer\vcpkg\buildtrees\libiconv\msys-x64-windows-3-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_acquire_msys.cmake:524 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_configure_make.cmake:291 (vcpkg_acquire_msys)
  ports/libiconv/portfile.cmake:28 (vcpkg_configure_make)
  scripts/ports.cmake:142 (include)


Error: Building package libiconv:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libiconv:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
I ensured that I'm using the latest portfiles.
The added the mentioned log file `msys-x64-windows-3-err.log`: [msys-x64-windows-3-err.log](https://github.com/microsoft/vcpkg/files/6317817/msys-x64-windows-3-err.log)
`

**Comment by autoantwort** [comment]
Looks like the downloaded file is broken. Maybe delete the folder `U:/Developer/vcpkg/downloads` and retry the installation.

**Comment by t-sharp** [comment]
Sorry, but that didn't helped.
I deleted the folder `U:/Developer/vcpkg/downloads` and tried to build again. The build still fails. The output from PowerShell is the following:
```
Computing installation plan...
The following packages will be built and installed:
    libiconv[core]:x64-windows -> 1.16#8
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\Tobias\AppData\Local\vcpkg\archives\26\26d2167072fc54a6fbc6f1af5a41842c0ebde6d6.zip
Starting package 1/1: libiconv:x64-windows
Building package libiconv[core]:x64-windows...
-- Downloading https://ftp.gnu.org/gnu/libiconv/libiconv-1.16.tar.gz -> libiconv-1.16.tar.gz...
-- Cleaning sources at U:/Developer/vcpkg/buildtrees/libiconv/src/1.16-de9101a181.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source U:/Developer/vcpkg/downloads/libiconv-1.16.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Using source at U:/Developer/vcpkg/buildtrees/libiconv/src/1.16-de9101a181.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Downloading https://repo.msys2.org/msys/x86_64/bash-4.4.023-2-x86_64.pkg.tar.xz -> msys-bash-4.4.023-2-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/autoconf-2.69-5-any.pkg.tar.xz -> msys-autoconf-2.69-5-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/diffutils-3.7-1-x86_64.pkg.tar.xz -> msys-diffutils-3.7-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/binutils-2.34-4-x86_64.pkg.tar.zst -> msys-binutils-2.34-4-x86_64.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/libtool-2.4.6-9-x86_64.pkg.tar.xz -> msys-libtool-2.4.6-9-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/file-5.39-1-x86_64.pkg.tar.zst -> msys-file-5.39-1-x86_64.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/zlib-1.2.11-1-x86_64.pkg.tar.xz -> msys-zlib-1.2.11-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/libbz2-1.0.8-2-x86_64.pkg.tar.xz -> msys-libbz2-1.0.8-2-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/coreutils-8.32-1-x86_64.pkg.tar.xz -> msys-coreutils-8.32-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/grep-3.0-2-x86_64.pkg.tar.xz -> msys-grep-3.0-2-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/sed-4.8-1-x86_64.pkg.tar.xz -> msys-sed-4.8-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/libpcre-8.44-1-x86_64.pkg.tar.xz -> msys-libpcre-8.44-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/m4-1.4.18-2-x86_64.pkg.tar.xz -> msys-m4-1.4.18-2-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/automake-wrapper-11-1-any.pkg.tar.xz -> msys-automake-wrapper-11-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/gawk-5.1.0-1-x86_64.pkg.tar.xz -> msys-gawk-5.1.0-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/mpfr-4.1.0-1-x86_64.pkg.tar.zst -> msys-mpfr-4.1.0-1-x86_64.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/gmp-6.2.0-1-x86_64.pkg.tar.xz -> msys-gmp-6.2.0-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/libreadline-8.0.004-1-x86_64.pkg.tar.xz -> msys-libreadline-8.0.004-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/ncurses-6.2-1-x86_64.pkg.tar.xz -> msys-ncurses-6.2-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/automake1.16-1.16.2-2-any.pkg.tar.zst -> msys-automake1.16-1.16.2-2-any.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/perl-5.32.0-2-x86_64.pkg.tar.zst -> msys-perl-5.32.0-2-x86_64.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/libcrypt-2.1-2-x86_64.pkg.tar.xz -> msys-libcrypt-2.1-2-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/pkg-config-0.29.2-4-x86_64.pkg.tar.zst -> msys-pkg-config-0.29.2-4-x86_64.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/make-4.3-1-x86_64.pkg.tar.xz -> msys-make-4.3-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/findutils-4.7.0-1-x86_64.pkg.tar.xz -> msys-findutils-4.7.0-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/libintl-0.19.8.1-1-x86_64.pkg.tar.xz -> msys-libintl-0.19.8.1-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/libiconv-1.16-2-x86_64.pkg.tar.zst -> msys-libiconv-1.16-2-x86_64.pkg.tar.zst...
-- Downloading https://repo.msys2.org/msys/x86_64/gcc-libs-9.3.0-1-x86_64.pkg.tar.xz -> msys-gcc-libs-9.3.0-1-x86_64.pkg.tar.xz...
-- Downloading https://repo.msys2.org/msys/x86_64/msys2-runtime-3.1.6-3-x86_64.pkg.tar.xz -> msys-msys2-runtime-3.1.6-3-x86_64.pkg.tar.xz...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" -E tar xzf U:/Developer/vcpkg/downloads/msys-binutils-2.34-4-x86_64.pkg.tar.zst
    Working Directory: U:/Developer/vcpkg/downloads/tools/msys2/d589ebc2087b2b2c.tmp
    Error code: 1
    See logs for more information:
      U:\Developer\vcpkg\buildtrees\libiconv\msys-x64-windows-3-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_acquire_msys.cmake:524 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_configure_make.cmake:291 (vcpkg_acquire_msys)
  ports/libiconv/portfile.cmake:28 (vcpkg_configure_make)
  scripts/ports.cmake:142 (include)


Error: Building package libiconv:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libiconv:x64-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
```

The text in the mentioned log file is as following:
```
CMake Error: Problem with archive_write_header(): Can't create '\\?\U:\Developer\vcpkg\downloads\tools\msys2\d589ebc2087b2b2c.tmp\usr\bin\ld.exe'
CMake Error: Current file: usr/bin/ld.exe
CMake Error: Problem extracting tar: U:/Developer/vcpkg/downloads/msys-binutils-2.34-4-x86_64.pkg.tar.zst
```

**Comment by autoantwort** [comment]
Maybe try this: #6059

**Comment by t-sharp** [comment]
No, sorry, that didn't work either.

I read the issue you linked and the UTF-8 beta in intl.cpl > Administrative > Change system locale...  was already unticked on my system. I also read [the issue](https://gitlab.kitware.com/cmake/cmake/-/issues/19102#note_558431) that was linked in the issue you mentioned and tried to change the codepage of PowerShell to 437 instead of the 65001 I previously used. But even the changed codepage resulted in the same error.

The error (both the log from the PowerShell and the content of the log file) remains in both cases the same (e. g. I have the same output).

**Comment by JackBoosY** [comment]
cc @cenit

**Comment by JackBoosY** [comment]
Maybe it's a permission issue.

**Comment by t-sharp** [comment]
I'm not sure if that's the case. I deactivated the controlled folder access in Windows Defender and tried again unsuccessfully (I thought that you meant this when you wrote "permission issue"). I even tried to delete the content of the entire folder `U:\Developer\vcpkg` and cloned `vcpkg` in that folder and tried to built `libiconv:x64-windows` again but still unsuccessfully. Then I did the same but in the path `U:\dev\vcpkg` but with the same result. After that, I tried the same again in the path `C:\dev\vcpkg` and it worked and I built successfully `libiconv:x64-windows`. I don't know why it worked - maybe it's the drive letter `U:` but I'm not sure.

I'm closing this issue because I found a workaround that works for me. Thank you for your help and your suggesstions.
